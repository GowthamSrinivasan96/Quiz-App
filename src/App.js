import React, { useState, useEffect } from 'react';
import './App.scss';
import './bootstrap.min.css';
import axios from 'axios';
import { func } from 'prop-types';
import { stat } from 'fs';
import Question from './Question';

function App() {
  const [timer,setTimer] = useState(60);
  const [qno,setQno] = useState([]);
  const [index,setIndex] = useState(0);
  const [crctAns,setCrctAns] = useState('');
  const [quizId,setQuizId] = useState(0);
  const [score,setScore] = useState(0);
  const [lastPage,setLastPage] = useState('');
  

  function shuffle(arra1) {
    var ctr = arra1.length, temp, index;
    while (ctr > 0) {
        index = Math.floor(Math.random() * ctr);
        ctr--;
        temp = arra1[ctr];
        arra1[ctr] = arra1[index];
        arra1[index] = temp;
    }
    return arra1;
  }

  useEffect(()=>{
    axios.get('https://opentdb.com/api.php?amount=10&category=9&difficulty=easy&type=multiple')
  .then(function (response) {
    const items =qno;
    response.data.results.map((result)=>{
      items.push(result);
      items[items.length-1].options = shuffle([result.correct_answer].concat(result.incorrect_answers));
    })
    setQno(items);
  })
  .catch(error => {
    console.log(error);
  })
  },[qno]);

  useEffect(()=>{
    let interval=null;
    if(timer > 0) {
      interval = setTimeout(() => {
        setTimer(timer => timer-1);
      },1000);
    }
    return () => clearTimeout(interval);
  });

  function checkAnswer(qInfo,selectedOption) {
   let currentQues = qno.find((question) => {
      return  question.question === qInfo.question
   })
   var element = document.getElementById(selectedOption);
   if(currentQues.correct_answer === selectedOption) {
     element.classList.add('success');
     setCrctAns('yes');
     setScore(score+1);
     setTimeout(()=> {setIndex(index+1);setCrctAns('')},1000);
   }
   else {
     setCrctAns('no');
     element.classList.add('failure');
   }
  }

  function resumeGame() {
    setQno([]);
    setQuizId(quizId+1);
    setIndex(0);
    if(lastPage === 'yes') {
      setLastPage('');
    }
    setTimer(60);

  }

  function nextQuestion() {
    setIndex(index+1);
    if(index === qno.length-1) {
       setLastPage('yes');
    }
    setCrctAns('');
  }

  return (
   <div className = "defaultBody">
     <div class="fluid-container login-container">
       <div className="header">
        <h2 style={{display:'inline-block'}}>Trivia-App </h2>
        <h3 style={{display:'inline-block',float:'right',paddingRight:35+'px'}}>Score {score}</h3>
        {timer > 0 && lastPage !== 'yes' ?<><h4 className="align-header">Questions {index+1} / {qno.length}</h4><h5 className="align-header">Remaining Time: {timer} sec(s)</h5></> :   <div className="endGameHeader"><h2> Game Over</h2><h2>Your Score {score}</h2><button className="btn-login" onClick={resumeGame} style={{float:'none'}}>Play Again</button></div>}
       </div>
        <div class=" login-center-row">
            <div class="form-container">
                <div class="js-enabled-panel">
                    <div class=" login-content">
                      <Question key={quizId} timer={timer} lastPage={lastPage} index={index} qno={qno} crctAns={crctAns} onclick={checkAnswer}/>
                    </div>
                </div>
                {qno.length && qno[index] &&  qno[index].options && qno[index].options.length && timer >0 ? 
                 (<button id="btnLogin" class="btn-login" data-cy="btn" type="submit" onClick={nextQuestion}>Continue</button>):''
                }
            </div>
        </div>
    </div>
   </div>
  );
}

export default App;
