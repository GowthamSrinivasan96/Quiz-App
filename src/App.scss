@import './variables.scss';
.defaultBody {
  padding-top: 40px;
  padding-bottom: 40px;
  height: $window-height;
  color:#78838d;
  position: relative;
  background: #d7d7d7 url('./3.0-fm-login-background-v2.jpg') repeat fixed;
  background-color: rgb(36, 37, 36);
  background-position: 66.5% 50%; /* position the background image so that the edge of the triangle lines up with the edge of the browser window */
  -webkit-background-size: 143%;
  -moz-background-size: 143%;
  -o-background-size: 143%;
  background-size: 143%;
  
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.login-container {
  height: 100%;
  width: 100%;
  max-width: 1000px;
  max-height: 701px;
  background-color:$form-bg-color;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  left: 50%;
}


.login-container * {
  box-sizing: unset;
}

.login-center-row {
  margin: 79px 71px 73px 103px;
}

@media (max-width: 768px) {
  .login-center-row {
      max-width: 450px;
  }
}

.login-copyright-notice {
  margin-top: 35px; /* Line up with logo text */
  color:$footer-notice-color;
  font-size: $copyright-font-size;
  line-height: 1.33;
}

.btn-login {
  border: 0;
  background:$btn-bg-color;
  color: $btn-color;
  font-size: $btn-font-size;
  border-radius: 10px;
  float: right;
  position: relative;
  top: 30px;
  padding: 6px 25px;
}

.form-container {
  margin-top: 60px;
}

.input-container {
  position: relative;
}

.login-footer {
  position: relative;
  text-align: center;
}

.banner-text-container {
  margin-top: 35px;
  width: 455px;
  display: inline-block;
}

.question-header {
    border:2px solid slategray;
    overflow: auto;
    height: 60px;
    padding: 10px 8px 10px 31px;
    background-color:#929192b0;
    font-size: 24px;
    color: black;
}

.flex-container {
  display: flex;
  flex-wrap: wrap;
}

.flex-container {
    div {
    background-color:$flex-container-div;
    margin: 10px;
    line-height: 75px;
    font-size: 24px;
  }
}
.header{
  padding: 20px 10px 0px 30px;
}

.align-header {
  text-align: center;
}

.success {
  border:2px solid green;
}

.failure {
  border:2px solid red;
}

.endGameHeader {
  text-align: center;
  position: fixed;
  top: 40%;
  left: 0;
  right: 0;
}

input[type=button]:focus {
  outline:0;
}