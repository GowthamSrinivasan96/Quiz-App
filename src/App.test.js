import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('renders Trivia-app', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Trivia-App/i);
  expect(linkElement).toBeInTheDocument();
});

it('matches the snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('App component', () => {
  it('Btn defined or not', () => {
    const wrapper  = shallow(<App />);
    expect(wrapper.find('input[data-cy="autoLogoutTime"]')).toBeDefined();
  });
});
