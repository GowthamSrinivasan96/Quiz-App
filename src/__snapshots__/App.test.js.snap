// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="defaultBody"
>
  <div
    class="fluid-container login-container"
  >
    <div
      className="header"
    >
      <h2
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        Trivia-App 
      </h2>
      <h3
        style={
          Object {
            "display": "inline-block",
            "float": "right",
            "paddingRight": "35px",
          }
        }
      >
        Score 
        0
      </h3>
      <h4
        className="align-header"
      >
        Questions 
        1
         / 
        0
      </h4>
      <h5
        className="align-header"
      >
        Remaining Time: 
        60
         sec(s)
      </h5>
    </div>
    <div
      class=" login-center-row"
    >
      <div
        class="form-container"
      >
        <div
          class="js-enabled-panel"
        >
          <div
            class=" login-content"
          >
            <form
              autocomplete="off"
              class="form-signin"
            >
              <div
                className="form-group question-header"
              >
                
              </div>
              <div
                className="flex-container"
              >
                
              </div>
            </form>
            <div
              class="login-footer"
            >
              <div
                class="banner-text-container"
              >
                <div
                  class="login-copyright-notice"
                >
                  
                </div>
              </div>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
